File: twoSum/twoSum.ts
function twoSum(arr: number[], target: number): number[] | null {
    const map = new Map<number,number>()
    for (let i = 0; i < arr.length; i++) {
      const x = target - arr[i];
      if(map.has(x)) return [i, map.get(x)!] 
      map.set(arr[i], i)
  }
    return null;
}

File: twoSum/twoSum.py
def two_sum(arr: list[int], target: int) -> list[int] | None:
    map = {}
    for i in range(len(arr)):
        x = target - arr[i]
        if x in map:
            return [map[x], i]
        map[arr[i]] = i
    return None

